src/python/zquantum/optimizers/grid_search.py

```diff
@@ -47,23 +48,20 @@ def minimize(
         min_value = None
         nfev = 0
 
+        if self.keep_value_history:
+            cost_function = recorder(cost_function)
```

Comment:
I think that right now there's no way to set recorder settings without changing the code, right?
Shouldn't we have an input parameter/class field or sth. like this?

Reply:
You are right. I didn't want this PR to get too large to digest, so I focused only on reproducing current functionalities. Adding recorder settings (i.e. save condition) will be proposed in another PR, with a corresponding change in the core's Optimizer interface.

Reply:
Sounds good 👍